src/datasets/packaged_modules/spark/spark.py

```diff
@@ -31,21 +32,37 @@ class SparkConfig(datasets.BuilderConfig):
     features: Optional[datasets.Features] = None
 
 
+def reorder_dataframe_by_partition(df: "pyspark.sql.DataFrame", new_partition_order: List[int]):
```

Comment:
Add a leading underscore to indicate that this function shouldn't be called from outside this file (_ reorder_dataframe_by_partition)